pkg/ottl/ottlfuncs/func_trim.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TrimArguments[K any] struct {
+	Target ottl.StringGetter[K]
```

Comment:
Lets add an optional parameter that allows specifying what to trim.  I believe it should be a string and the converter will trim that string from both sides of the target string.